ci-operator/step-registry/idp/external-oidc/keycloak/aks/server/idp-external-oidc-keycloak-aks-server-commands.sh

```diff
@@ -0,0 +1,395 @@
+    password=$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true)
+    TEST_USERS+="${username}:${password},"
+done
+# The users will be mounted into keycloak pods and the real test users will be created there
+TEST_USERS=${TEST_USERS%?}
+
+# Install NGINX Ingress Controller
+/tmp/linux-amd64/helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+/tmp/linux-amd64/helm repo update
+/tmp/linux-amd64/helm install ingress-nginx ingress-nginx/ingress-nginx \
+  --namespace ingress-nginx \
+  --create-namespace \
+  --set controller.service.annotations."service\.beta\.kubernetes\.io/azure-load-balancer-health-probe-request-path"=/healthz \
+  --wait
+
+# Install cert-manager
+oc create namespace cert-manager || true
+oc apply -f https://github.com/jetstack/cert-manager/releases/download/v1.13.0/cert-manager.yaml
+oc wait --for=condition=ready pod -l app.kubernetes.io/instance=cert-manager -n cert-manager --timeout=300s
+# Create Let's Encrypt ClusterIssuer
+cat <<EOF | oc apply -f -
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-prod
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: openshift-qe@redhat.com
+    privateKeySecretRef:
+      name: letsencrypt-prod
+    solvers:
+    - http01:
+        ingress:
+          class: nginx
+EOF
+oc wait --for=condition=ready clusterissuer letsencrypt-prod --timeout=120s
+
+# Set Keycloak repo
+/tmp/linux-amd64/helm repo add bitnami https://charts.bitnami.com/bitnami
+/tmp/linux-amd64/helm repo update
+
+# Create Keycloak settings file
+cat > /tmp/keycloak-values.yaml << EOF
+nameOverride: ""
+fullnameOverride: ""
+namespace: keycloak
+
+auth:
+  adminUser: ${KEYCLOAK_ADMIN_TEST_USER}
+  adminPassword: "${KEYCLOAK_ADMIN_TEST_PASSWORD}"
+image:
+  registry: docker.io
+  repository: bitnamilegacy/keycloak
+  tag: 26.3.3-debian-12-r0
+  pullPolicy: IfNotPresent
+
+securityContext:
+  enabled: true
+  fsGroup: 1001
+  runAsUser: 1001
+  runAsNonRoot: true
+
+envFrom:
+  - configMapRef:
+      name: "keycloak-env-vars"
+  - secretRef:
+      name: "keycloak"
+
+extraVolumes:
+  - name: setup-scripts
+    configMap:
+      name: "keycloak-setup-scripts"
+      defaultMode: 0755
+
+extraVolumeMounts:
+  - name: setup-scripts
+    mountPath: /opt/bitnami/keycloak/setup-scripts/
+    readOnly: true
+
+lifecycleHooks:
+  postStart:
+    exec:
+      command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+          set -euo pipefail
+
+          echo "Waiting for Keycloak server to be ready..."
+          timeout 10m bash -c 'while true; do
+            if curl -s http://localhost:${KC_HTTP_PORT:-8080}/health/ready; then
+              echo "Keycloak server is ready. Starting initialization..."
+              break
+            fi
+            sleep 5
+          done' || {
+            echo "ERROR: Timeout waiting for Keycloak server"
+            exit 1
+          }
```

Comment:
If there is failure in these lines, it is hard to debug where the failure is. Because we do not have way to examine the echo outputs of these lines. The `oc logs` way doesn't examine these outputs. The only way is: we have to put all outputs to /tmp/postStart.log and examine postStart.log instead.